homeassistant/components/hassio/__init__.py

```diff
@@ -388,8 +388,12 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:  # noqa:
             if not user.is_admin:
                 await hass.auth.async_update_user(user, group_ids=[GROUP_ID_ADMIN])
 
+            # Migrate old name
+            if user.name == "Hass.io":
+                await hass.auth.async_update_user(user, name="Supervisor")
+
     if refresh_token is None:
-        user = await hass.auth.async_create_system_user("Hass.io", [GROUP_ID_ADMIN])
+        user = await hass.auth.async_create_system_user("Supervisor", [GROUP_ID_ADMIN])
```

Comment:
Do we have a test for the new user name? If not, should we add that?

Reply:
It's just the name used in the frontend?

Reply:
Yes, but we obviously care about it. We can just add an assert on this line:
https://github.com/home-assistant/core/blob/3276666457c985738ffb255fd1e959c4d58433db/tests/components/hassio/test_init.py#L179

Reply:
<https://github.com/home-assistant/core/pull/51771/files#diff-d3226201433d460b6c7c02902688abb43c4c61c5b2d0dd11c2fcd0995f808443R225> ?

Reply:
That tests when an existing user name is migrated. It doesn't test this line where we create a new user with a name.